### ISSUE TITLE
Archived label visual bug in labels select

### ISSUE THREAD

**Issue by stuzer05** [go-gitea/gitea#27466]
### Description

When you have archived non-exclusive label between exclusive label groups the double divider is shown in the place of archived label

### Gitea Version

trunk

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

![image](https://github.com/go-gitea/gitea/assets/23124682/264de0f0-e129-487e-8e19-204164b64679)
![image](https://github.com/go-gitea/gitea/assets/23124682/8ecd5dc2-0d08-48b4-8607-5552c0145885)

### How are you running Gitea?

docker

### Database

None

**Comment by neel1996** [comment]
The issue is due to the divider being rendered unconditionally

This has been fixed as part of #27668

**Comment by delvh** [comment]
I think the easiest approach to completely fix the problem is by refactoring the code completely.
As far as I know, @puni9869 is already working on sorting archived labels to the bottom of the search results, which will fix this problem simultaneously and much less error-prone.

**Comment by puni9869** [comment]
Y

> The issue is due to the divider being rendered unconditionally
> 
> This has been fixed as part of #27668

This is known issue.

**Comment by wxiaoguang** [comment]
Does the problem still exist in 1.23?

**Comment by stuzer05** [comment]
> Does the problem still exist in 1.23?

It does
![image](https://github.com/user-attachments/assets/6b9d8338-4660-4f47-873b-392487fc1d1d)

Gitea 6bd70d4d87262e3bf423aa8a598cfdcf3aae4008

**Comment by stuzer05** [comment]
repo labels ![image](https://github.com/user-attachments/assets/2506bdcb-dbc1-4717-ab49-09f2b947a993)
org labels ![image](https://github.com/user-attachments/assets/62ad329a-f548-462d-903b-f3d5e3c535b3)

**Comment by wxiaoguang** [comment]
The problem is that `tw-hidden` doesn't work with the dropdown divider. It can't be fixed by template patches.

**Comment by stuzer05** [comment]
> The problem is that `tw-hidden` doesn't work with the dropdown divider. It can't be fixed by template patches.

I don't know how labels work currently, but I'd imagine backend would for a merged list of labels aka. map[map[int64, string]] and frontend would just print that nested grouped map. With this logic, but would never happen

**Comment by wxiaoguang** [comment]
> > The problem is that `tw-hidden` doesn't work with the dropdown divider. It can't be fixed by template patches.
> 
> I don't know how labels work currently, but I'd imagine backend would for a merged list of labels aka. map[map[int64, string]] and frontend would just print that nested grouped map. With this logic, but would never happen

It's not that easy. Frontend dropdown needs to "filter" the list. So it also needs to handle such case:

```
item1
divider
item2 (not matched the input and will be hidden by frontend)
divider (this will also be hidden)
item3
```

Actually the dropdown could already handles such case, but it is unable to mix these 2 different usages if item2 is hidden by other methods but not by the filter.

**Comment by stuzer05** [comment]
> if item2 is hidden by other methods

Shouldn't those items that are hidden by other methods be non-existent for dropdown? Where am I missing?

**Comment by wxiaoguang** [comment]
> > if item2 is hidden by other methods
> 
> Shouldn't those items that are hidden by other methods be non-existent for dropdown? Where am I missing?

* Hidden by "archive": `class="tw-hidden"`
* Hidden by "filter": `class="filtered"`
* And `filtered` can't be used by "Hidden by 'archive'" (anther long story, you could take a try to see what would happen)

**Comment by stuzer05** [comment]
Maybe I fixed it, try this patch

![image](https://github.com/user-attachments/assets/14158136-3968-4510-83c9-d98208ec14cd)

```diff
diff --git a/templates/repo/issue/sidebar/label_list.tmpl b/templates/repo/issue/sidebar/label_list.tmpl
index fb8f1a667..1d34cf7a7 100644
--- a/templates/repo/issue/sidebar/label_list.tmpl
+++ b/templates/repo/issue/sidebar/label_list.tmpl
@@ -21,5 +21,7 @@
 					{{$previousExclusiveScope := "_no_scope"}}
-					{{range $data.RepoLabels}}
+					{{$previousHadDivider := false}}
+					{{range $data.AllLabels}}
 						{{$exclusiveScope := .ExclusiveScope}}
-						{{if and (ne $previousExclusiveScope "_no_scope") (ne $previousExclusiveScope $exclusiveScope)}}
+						{{if and (ne $previousExclusiveScope "_no_scope") (ne $previousExclusiveScope $exclusiveScope) (ne $previousHadDivider true)}}
+							{{$previousHadDivider = true}}
 							<div class="divider"></div>
@@ -27,13 +29,6 @@
 						{{$previousExclusiveScope = $exclusiveScope}}
-						{{template "repo/issue/sidebar/label_list_item" dict "Label" .}}
-					{{end}}
-					{{if and $data.RepoLabels $data.OrgLabels}}<div class="divider"></div>{{end}}
-					{{$previousExclusiveScope = "_no_scope"}}
-					{{range $data.OrgLabels}}
-						{{$exclusiveScope := .ExclusiveScope}}
-						{{if and (ne $previousExclusiveScope "_no_scope") (ne $previousExclusiveScope $exclusiveScope)}}
-							<div class="divider"></div>
+						{{if not .IsArchived}}
+							{{$previousHadDivider = false}}
+							{{template "repo/issue/sidebar/label_list_item" dict "Label" .}}
 						{{end}}
-						{{$previousExclusiveScope = $exclusiveScope}}
-						{{template "repo/issue/sidebar/label_list_item" dict "Label" .}}
 					{{end}}

```

**Comment by stuzer05** [comment]
maybe this is better

```
{{if not .IsArchived}}
	{{$previousHadDivider = false}}
{{end}}
{{template "repo/issue/sidebar/label_list_item" dict "Label" .}}
```

**Comment by stuzer05** [comment]
Tested, this patch fixes all cases, including archived label selected and not

```diff
diff --git a/templates/repo/issue/sidebar/label_list.tmpl b/templates/repo/issue/sidebar/label_list.tmpl
index fb8f1a667..8fd0ff5dd 100644
--- a/templates/repo/issue/sidebar/label_list.tmpl
+++ b/templates/repo/issue/sidebar/label_list.tmpl
@@ -21,19 +21,11 @@
 					{{$previousExclusiveScope := "_no_scope"}}
-					{{range $data.RepoLabels}}
-						{{$exclusiveScope := .ExclusiveScope}}
-						{{if and (ne $previousExclusiveScope "_no_scope") (ne $previousExclusiveScope $exclusiveScope)}}
-							<div class="divider"></div>
+					{{range $data.AllLabels}}
+						{{if or (not .IsArchived) (.IsChecked)}}
+							{{$exclusiveScope := .ExclusiveScope}}
+							{{if and (ne $previousExclusiveScope "_no_scope") (ne $previousExclusiveScope $exclusiveScope)}}
+								<div class="divider"></div>
+							{{end}}
+							{{$previousExclusiveScope = $exclusiveScope}}
+							{{template "repo/issue/sidebar/label_list_item" dict "Label" .}}
 						{{end}}
-						{{$previousExclusiveScope = $exclusiveScope}}
-						{{template "repo/issue/sidebar/label_list_item" dict "Label" .}}
-					{{end}}
-					{{if and $data.RepoLabels $data.OrgLabels}}<div class="divider"></div>{{end}}
-					{{$previousExclusiveScope = "_no_scope"}}
-					{{range $data.OrgLabels}}
-						{{$exclusiveScope := .ExclusiveScope}}
-						{{if and (ne $previousExclusiveScope "_no_scope") (ne $previousExclusiveScope $exclusiveScope)}}
-							<div class="divider"></div>
-						{{end}}
-						{{$previousExclusiveScope = $exclusiveScope}}
-						{{template "repo/issue/sidebar/label_list_item" dict "Label" .}}
 					{{end}}

```

**Comment by wxiaoguang** [comment]
No, it won't work.

See this:  Fix duplicate dropdown dividers #32760

**Comment by stuzer05** [comment]
> No, it won't work.
> 
> See this: Fix duplicate dropdown dividers #32760

But it does, see screenshots https://github.com/go-gitea/gitea/pull/32758

**Comment by puni9869** [comment]
@stuzer05 Thanks for the taking this forward. Now a days I am packed up. This was in my todo list from last year. But unfortunately I could not able to fix it.
